pkgs/phps.nix

```diff
@@ -43,9 +54,14 @@ let
 
               "--enable-libxml"
               "--with-libxml-dir=${prev.libxml2.dev}"
+              (
+                if prev.lib.versionAtLeast args.version "7.3"
+                then "--with-pcre-regex=${prev.pcre2.dev}"
+                else "--with-pcre-regex=${prev.pcre.dev}"
+              )
             ]
-            ++ prev.lib.optionals (prev.lib.versionAtLeast args.version "7.3") [
-              "--with-pcre-regex=${prev.pcre2.dev}"
+            ++ prev.lib.optionals (prev.lib.versionOlder args.version "7.3") [
+              "PCRE_LIBDIR=${prev.pcre}"
```

Comment:
I would try just

```suggestion
               "--with-pcre-regex=${prev.pcre.dev}"
              "PCRE_LIBDIR=${prev.pcre}"
```

Reply:
@HelloWorld017 What about this suggestion, is this something valid ?

Reply:
I think we should add `"--with-pcre-regex=${prev.pcre2.dev}"` for php7.3 so it might not be applicable. But in the point of the view of readability, it might be better to be something like this:

```nix
++ prev.lib.optionals (prev.lib.versions.majorMinor args.version == "7.3") [
  "--with-pcre-regex=${prev.pcre2.dev}"
]
++ prev.lib.optionals (prev.lib.versionOlder args.version "7.3") [
  "--with-pcre-regex=${prev.pcre.dev}"
  "PCRE_LIBDIR=${prev.pcre}"
];
```

Reply:
Oh, I see now, the pkg-config check is wrapped in `if test "$PHP_PCRE_REGEX" = "/usr"; then` for some reason. I would prefer

```
++ prev.lib.optionals (prev.lib.versions.majorMinor args.version == "7.3") [
  # Force use of pkg-config.
  # https://github.com/php/php-src/blob/php-7.3.33/ext/pcre/config0.m4#L14
  "--with-pcre-regex=/usr"
]
++ prev.lib.optionals (prev.lib.versionOlder args.version "7.3") [
  # Only PCRE 1 supported and no pkg-config.
  "--with-pcre-regex=${prev.pcre.dev}"
  "PCRE_LIBDIR=${prev.pcre}"
];
```

Reply:
I pushed those changes.